src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+  memo,
+  useRef,
+  useState,
+  useEffect,
+  useContext,
+} from 'react';
+
+import { Action } from '../../types';
+import { TodosContext } from '../TodosProvider';
+
+export const NewTodo: React.FC = memo(() => {
+  const { dispatch } = useContext(TodosContext);
```

Comment:
You can create a hook for your contexts and in this case, you don't need to pass context every time:
```javascript
const useTodos = () => useContext(TodosContext);
```